pkg/controller/control.go

```diff
@@ -301,6 +302,9 @@ func (c *Controller) finishExpansion(expanded bool, size int64) {
 	c.Unlock()
```

Comment:
```suggestion
	defer c.Unlock()
```

Reply:
BTW, we need to backport the fix to v1.3.x and v1.2.x.

Reply:
Sure. Will do it after the expansion PR merge.